frontend/app/src/composables/balances/token-detection.ts

```diff
@@ -1,57 +1,70 @@
 import { type MaybeRef } from '@vueuse/core';
-import { type Blockchain } from '@rotki/common/lib/blockchain';
+import { Blockchain } from '@rotki/common/lib/blockchain';
 import { type Ref } from 'vue';
-import { useBlockchainBalancesStore } from '@/store/blockchain/balances';
 import { useBlockchainTokensStore } from '@/store/blockchain/tokens';
 import { useTasks } from '@/store/tasks';
 import { TaskType } from '@/types/task-type';
 import { assert } from '@/utils/assertions';
 import { isTokenChain } from '@/types/blockchain/chains';
+import { useEthAccountsStore } from '@/store/blockchain/accounts/eth';
+import { useChainsAccountsStore } from '@/store/blockchain/accounts/chains';
 
 export const useTokenDetection = (
   chain: Ref<Blockchain>,
   accountAddress: MaybeRef<string | null> = null
 ) => {
   const { isTaskRunning } = useTasks();
-  const { fetchBlockchainBalances } = useBlockchainBalancesStore();
-  const { getEthDetectedTokensInfo, fetchDetectedTokens } =
-    useBlockchainTokensStore();
+  const {
+    getEthDetectedTokensInfo,
+    fetchDetectedTokens: fetchDetectedTokensCaller
+  } = useBlockchainTokensStore();
+
+  const { ethAddresses } = storeToRefs(useEthAccountsStore());
+  const { optimismAddresses } = storeToRefs(useChainsAccountsStore());
 
   const detectingTokens = computed<boolean>(() => {
     const address = get(accountAddress);
+    const blockchain = get(chain);
     return get(
-      isTaskRunning(TaskType.FETCH_DETECTED_TOKENS, address ? { address } : {})
+      isTaskRunning(TaskType.FETCH_DETECTED_TOKENS, {
+        chain: blockchain,
+        ...(address ? { address } : {})
+      })
     );
   });
 
   const detectedTokens = getEthDetectedTokensInfo(chain, accountAddress);
 
-  const fetchDetectedTokensAndQueryBalance = async (address: string) => {
+  const fetchDetectedTokens = async (address: string) => {
     const blockchain = get(chain);
     assert(isTokenChain(blockchain));
-    await fetchDetectedTokens(blockchain, address);
-    await fetchBlockchainBalances({
-      blockchain,
-      ignoreCache: true
-    });
```

Comment:
I remove the fetch balances part here, because it can cause the same bug we already resolved. 
https://discord.com/channels/657906918408585217/683245846070165504/1044908140958842970

Reply:
The code from `frontend/app/src/store/blockchain/tokens.ts` should be enough to fetch balances only if no tokens detection that running anymore.

`
watch(isEthDetecting, async (isDetecting, wasDetecting) => {
    if (wasDetecting && !isDetecting) {
      await fetchBlockchainBalances({
        blockchain: Blockchain.ETH,
        ignoreCache: true
      });
    }
  });
  watch(isOptimismDetecting, async (isDetecting, wasDetecting) => {
    if (wasDetecting && !isDetecting) {
      await fetchBlockchainBalances({
        blockchain: Blockchain.OPTIMISM,
        ignoreCache: true
      });
    }
  });
`